### PR TITLE
Bong refactor and water vapour removal

### PR DESCRIPTION
**Pull request by tf-4**
## About The Pull Request

Brings bong code more up to standard with early returns, autodocs, balloon alerts, etc.
Removes the water spill from the lungbuster, gives both bongs a fake vapour effect (Effect taken from https://github.com/hrzntal/horizon/pull/637).
The lungbuster has a range of 7, demonstrated, the normal bong a range of 2, not demonstrated.

Would be nice to have the false smoke as a component or something but that's a PR for another day.

image

## How This Contributes To The Skyrat Roleplay Experience

Having people sit in halls, escape shuttles, etc, and fill them with a slipping hazard because "haha dude, weed" is not enjoyable. At least with a canister of H2O there's a reasonable chance for someone to be able to take it.

## Changelog
:cl:
del: The lungbuster no longer spews (real) water on use. It now has a fake water vapour effect.
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/bongs/code/bong.dm [hunk 1]
```diff
@@ -23,38 +39,40 @@
...
-			if(bonghits > 0)
-				light(lighting_text)
-				name = "lit [initial(name)]"
-			else
-				to_chat(user, span_warning("There is nothing to smoke!"))
-		else
+		var/lighting_text = used_item.ignition_effect(src, user)
+		if(!lighting_text)
+			return ..()
+		if(!(bong_hits > 0))
```
Comment (Zonespace27) [inline modular_skyrat/modules/bongs/code/bong.dm]: wouldn't this work as `if(bong_hits <= 0)`?

File: modular_skyrat/modules/bongs/code/bong.dm [hunk 2]
```diff
@@ -63,36 +81,47 @@
...
-					to_chat(hit_mob, span_notice("Your [name] goes out."))
-					lit = FALSE
-					packeditem = FALSE
-					icon_state = icon_off
-					inhand_icon_state = icon_off
-					name = "[initial(name)]"
-					reagents.clear_reagents() //just to make sure
+	if(!packed_item || !lit)
+		return
+	//if(hit_mob == user)
```
Comment (Zonespace27) [inline modular_skyrat/modules/bongs/code/bong.dm]: remove commented out code

File: modular_skyrat/modules/bongs/code/bong.dm [hunk 3]
```diff
@@ -63,36 +81,47 @@
...
+	hit_mob.visible_message(span_notice("[user] starts [hit_mob == user ? "taking a hit from [src]." : "forcing [hit_mob] to take a hit from [src]!"]"), hit_mob == user ? span_notice("You start taking a hit from [src].") : span_userdanger("[user] starts forcing you to take a hit from [src]!"))
+	//else
+		//hit_mob.visible_message(span_notice("[user] starts forcing [hit_mob] to take a hit from [src]!"), span_userdanger("[user] starts forcing you to take a hit from [src]!"))
+	playsound(src, 'sound/chemistry/heatdam.ogg', 50, TRUE)
+	if(!do_after(user,40))
```
Comment (Zonespace27) [inline modular_skyrat/modules/bongs/code/bong.dm]:
```suggestion
	if(!do_after(user, 40))
```

File: modular_skyrat/modules/bongs/code/bong.dm (hunk 2 above)
Comment (tf-4) [reply modular_skyrat/modules/bongs/code/bong.dm]: oh oops forgot to remove that

File: modular_skyrat/modules/bongs/code/bong.dm [hunk 4]
```diff
@@ -63,36 +81,47 @@
...
-					icon_state = icon_off
-					inhand_icon_state = icon_off
-					name = "[initial(name)]"
-					reagents.clear_reagents() //just to make sure
+	if(!packed_item || !lit)
+		return
+	//if(hit_mob == user)
+	hit_mob.visible_message(span_notice("[user] starts [hit_mob == user ? "taking a hit from [src]." : "forcing [hit_mob] to take a hit from [src]!"]"), hit_mob == user ? span_notice("You start taking a hit from [src].") : span_userdanger("[user] starts forcing you to take a hit from [src]!"))
+	//else
+		//hit_mob.visible_message(span_notice("[user] starts forcing [hit_mob] to take a hit from [src]!"), span_userdanger("[user] starts forcing you to take a hit from [src]!"))
```
Comment (tf-4) [inline modular_skyrat/modules/bongs/code/bong.dm]:
```suggestion
	hit_mob.visible_message(span_notice("[user] starts [hit_mob == user ? "taking a hit from [src]." : "forcing [hit_mob] to take a hit from [src]!"]"), hit_mob == user ? span_notice("You start taking a hit from [src].") : span_userdanger("[user] starts forcing you to take a hit from [src]!"))
```

File: modular_skyrat/modules/bongs/code/bong.dm (hunk 1 above)
Comment (tf-4) [inline modular_skyrat/modules/bongs/code/bong.dm]:
```suggestion
		if(bong_hits <= 0)
```

**Review by Gandalf2k15 — APPROVED** [review]

**Pull request merged**